Tests/TuistGeneratorTests/Generator/BuildPhaseGeneratorTests.swift

```diff
@@ -385,6 +385,49 @@ final class BuildPhaseGeneratorTests: TuistUnitTestCase {
         let pbxBuildFile: PBXBuildFile? = pbxBuildPhase?.files?.first
         XCTAssertEqual(pbxBuildFile?.file, fileElement)
     }
+    
+    func test_generateResourcesBuildPhase_whenContainsResourcesTags() throws {
+        // Given
+        let temporaryPath = try self.temporaryPath()
+        let resources: [FileElement] = [.file(path: "/file.type", tags: ["fileTag"]),
+                                        .folderReference(path: "/folder", tags: ["folderTag"])]
+        let target = Target.test(resources: resources)
+        let fileElements = ProjectFileElements()
+        let pbxproj = PBXProj()
+        
+        let fileElement = PBXFileReference()
+        let folderElement = PBXFileReference()
+        pbxproj.add(object: fileElement)
+        pbxproj.add(object: folderElement)
+        fileElements.elements["/file.type"] = fileElement
+        fileElements.elements["/folder"] = folderElement
+        
+        let expectedFileSettings: [String: AnyHashable] = ["ASSET_TAGS": resources.first?.tags]
+        let expectedFolderSettings: [String: AnyHashable] = ["ASSET_TAGS": resources[1].tags]
+
+        let nativeTarget = PBXNativeTarget(name: "Test")
+
+        let graph = ValueGraph.test(path: temporaryPath)
+        let graphTraverser = ValueGraphTraverser(graph: graph)
+
+        // When
+        try subject.generateResourcesBuildPhase(path: "/path",
+                                                target: target,
+                                                graphTraverser: graphTraverser,
+                                                pbxTarget: nativeTarget,
+                                                fileElements: fileElements,
+                                                pbxproj: pbxproj)
+
+        // Then
+        let pbxBuildPhase: PBXBuildPhase? = nativeTarget.buildPhases.first
+        XCTAssertNotNil(pbxBuildPhase)
+        XCTAssertTrue(pbxBuildPhase is PBXResourcesBuildPhase)
+        
+        let fileSettings = pbxBuildPhase?.files?.first?.settings as? [String: AnyHashable]
+        let folderSettings = pbxBuildPhase?.files?[1].settings as? [String: AnyHashable]
+        XCTAssertEqual(fileSettings, expectedFileSettings)
+        XCTAssertEqual(folderSettings, expectedFolderSettings)
```

Comment:
minor: An slightly simpler alternate we could leverage here is array comparisons, this ensures order, content and count.

```swift
let resourceBuildPhase = try XCTUnwrap(nativeTarget.buildPhases.first as? PBXResourcesBuildPhase)
let allFileSettings = resourceBuildPhase.files?.map { $0.settings as? [String: AnyHashable] }
XCTAssertEqual(allFileSettings, [
    ["ASSET_TAGS": ["fileTag"]],
    ["ASSET_TAGS": ["folderTag"]],
])
```